### PR TITLE
Use golangci lint action

### PR DESCRIPTION
**Pull request by Smolevich**
- Add `.golangci.yml` as config
- Add using `working-directory`
- Update `actions/checkout@v1` on `actions/checkout@v2`
- Add new path for triggers ci test_backend

I propose if this [issue](https://github.com/matoous/golangci-lint-action/issues/14) will be solved  use action for linting

### CODE REVIEW

File: .github/workflows/ci-test-backend.yml [hunk 1]
```diff
@@ -50,22 +52,19 @@ jobs:
           curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $GITHUB_WORKSPACE v1.20.0
           go get -u github.com/mattn/goveralls
 
-      - name: run backend linters
+      - name: run backend linters for backend
```
Comment (paskal) [inline .github/workflows/ci-test-backend.yml]: Looks unnecessary.

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+run:
```
Comment (paskal) [inline .golangci.yml]: Could you you switch docker image to it as well? Otherwise we’ll have two different configs for two places.

Comment (Smolevich) [reply .golangci.yml]: I change in Dockerfile

**Comment by umputun** [comment]
I'd suggest moving .golangci.yml to the `backend` directory

File: .github/workflows/ci-test-backend.yml (hunk 1 above)
Comment (paskal) [reply .github/workflows/ci-test-backend.yml]: I mean, it had “backend” word twice. Only one can live:)

Comment (Smolevich) [reply .github/workflows/ci-test-backend.yml]: Remove `backend` in end

**Comment by Smolevich** [comment]
> I'd suggest moving .golangci.yml to the `backend` directory

I moved `.golangci.yml` to the `backend` directory

File: .github/workflows/ci-test-backend.yml [hunk 3]
```diff
@@ -51,21 +53,17 @@ jobs:
           go get -u github.com/mattn/goveralls
 
       - name: run backend linters
+        run: $GITHUB_WORKSPACE/golangci-lint run --config .golangci.yml ./...
+        working-directory: backend
+
+      - name: run linters for examples
         run: |
```
Comment (paskal) [inline .github/workflows/ci-test-backend.yml]: This one can be converted to single line as well.

File: .github/workflows/ci-test-backend.yml [hunk 4]
```diff
@@ -51,21 +53,17 @@ jobs:
...
-          cd _example/memory_store
-          $GITHUB_WORKSPACE/golangci-lint run --out-format=tab --disable-all --tests=false --enable=unconvert \
-            --enable=megacheck --enable=structcheck --enable=gas --enable=gocyclo --enable=dupl --enable=misspell \
-            --enable=unparam --enable=varcheck --enable=deadcode --enable=typecheck \
-            --enable=ineffassign --enable=varcheck ./... ;
+          $GITHUB_WORKSPACE/golangci-lint run --config ${GITHUB_WORKSPACE}/backend/.golangci.yml ./...
+        working-directory: backend/_example/memory_store
 
       - name: submit coverage
         run: |
```
Comment (paskal) [inline .github/workflows/ci-test-backend.yml]: Can be single line as well.

Comment (Smolevich) [reply .github/workflows/ci-test-backend.yml]: ready

File: .github/workflows/ci-test-backend.yml (hunk 3 above)
Comment (Smolevich) [reply .github/workflows/ci-test-backend.yml]: ready

File: backend/.golangci.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    - gocyclo
+    - dupl
+    - misspell
+    - unparam
+    - varcheck
+    - deadcode
+    - typecheck
+    - ineffassign
+    - varcheck
+  disable-all: false
```
Comment (paskal) [inline backend/.golangci.yml]: I believe it's `true` in original. Other than that, LGTM.

Comment (Smolevich) [reply backend/.golangci.yml]: changed

**Comment by Smolevich** [comment]
@umputun, i received email notification about failed drone build by merge to master, but there isn't link on build in email. Is it really?

**Comment by umputun** [comment]
it failed to build a docker image on the private drone.  This is the error:

36sht-202001-13115111-9pknw

If you run docker build locally without SKIP tests flag you will see it too. I think you missed copying `.golangci.yml` to the right place in Dockerfile

**Comment by umputun** [comment]
I have it fixed in https://github.com/umputun/remark/commit/3c12fde0f87ff5454c8de63b0801a975ffaf59d4, docker build runs from `app` directory

**Pull request merged**